native/src/modules/endpoint/sagas/loadCityContent.js

```diff
@@ -54,74 +54,88 @@ export default function * loadCityContent (
   const shouldUpdate = criterion.shouldUpdate(lastUpdate)
   console.debug('City content should be refreshed: ', shouldUpdate)
 
-  const cities = yield call(loadCities, dataContainer, shouldUpdate) // Refresh for flags like eventsEnabled necessary
-  const cityModel = cities.find(city => city.code === newCity)
-  if (!cityModel) {
-    throw new Error(`City '${newCity}' was not found.`)
-  }
+  // Temporarily set lastUpdate to "now" to hinder other threads from trying to update content and
+  // resources at the same time. This kind of serves as a lock.
+  yield call(dataContainer.setLastUpdate, newCity, newLanguage, moment())
+
+  try {
```

Comment:
Yes.

Reply:
Actually, despite not remembering that the `finally` block existed, I was right to put it in the `catch` block, because we only want so reset `lastUpdate` to the old date if an error occurred. `finally` is always executed.